docs/manual/returns-receiver-checker.tex

```diff
@@ -0,0 +1,120 @@
+\htmlhr
+\chapterAndLabel{Returns Receiver Checker}{returns-receiver-checker}
+
+The Returns Receiver Checker enables documenting and checking that a method
+returns its receiver (i.e., the \<this> parameter).
+
+There are two ways to run the Returns Receiver Checker.
+\begin{itemize}
+\item
+Typically, it is automatically run by another checker.
+\item
+Alternately, you can run just the Returns Receiver Checker, by
+supplying the following command-line options to javac:
+\code{-processor org.checkerframework.common.returnsreceiver.ReturnsReceiverChecker}
+\end{itemize}
+
+
+\sectionAndLabel{Annotations}{returns-receiver-checker-annotations}
+
+The qualifier \refqualclass{common/returnsreceiver/qual}{This} on the return
+type of a method indicates that the method returns its receiver.  Methods
+that return their receiver are common in so-called ``fluent'' APIs.  Here
+is an example:
+
+\begin{Verbatim}
+class MyBuilder {
+  @This MyBuilder setName(String name) {
+    this.name = name;
+    return this;
+  }
+}
+\end{Verbatim}
+
+An \refqualclass{common/returnsreceiver/qual}{This} annotation can only be
+written on the return type of a method or in a downcast.
+
+As is standard, the Returns Receiver Checker has a top qualifier,
+\refqualclass{common/returnsreceiver/qual}{UnknownThis}, and a bottom qualifier,
+\refqualclass{common/returnsreceiver/qual}{BottomThis}.
+Programmers rarely need to write these annotations.
+
+Here are additional details.
+\refqualclass{common/returnsreceiver/qual}{This}
+is a polymorphic qualifier rather than a regular type qualifier.
+When the Returns Receiver Checker observes an
+\refqualclass{common/returnsreceiver/qual}{This} annotation on a method return type,
+it adds \refqualclass{common/returnsreceiver/qual}{This} to
+the method's receiver type. The presence of the polymorphic annotation
+\refqualclass{common/returnsreceiver/qual}{This} on both the method's return and
+receiver type forces their type qualifiers to be \emph{equal}.  Hence, the
+method will only pass the type checker if it returns its receiver argument,
+achieving the desired checking.
```

Comment:
Do you also want to add a sentence about the method caller perspective? The polymorphic annotation gets resolved to `UnknownThis` and the type on the caller's side doesn't really matter. But maybe that's too much detail.
It's just a very different way how this is using the polymorphic qualifier, so it might confuse people that understand that.

Reply:
Yes, I think this would be too much detail.  Hopefully what is written is enough for the user.